### ISSUE TITLE
[FEATURE] - Rewrite statistics

### ISSUE THREAD

**Issue by NobodyForNothing** [NobodyForNothing/blood-pressure-monitor-fl#179]
### What you want:

- [x] Redesign the statistics page so it doesn't have distinct borders
- [x] remove smallEdges property
- [x] rewrite statistics page code (testable, new style)
- [ ] add more useful metrics
- [ ] Time-Resolved Metrics could include more options on what to display
- [ ] ability to choose own metrics, move them, and create your own (this one should be done after #164 and #165 are done as it creates a lot of additional complexity)

### Why you want it:

Currently, the statistics page is only a small thought and doesn't live up to its potential. It's neither fast nor something I like to look at. 

In its current form it isn't really useful to anyone and people export the measurements to different apps for further analysis.

**Comment by xandro0777** [comment]
Blood pressure values are often very different at different times of the day so it should be possible to break down the statistics to consider measurements at similar times of day.. eg morning, noon, evening separately.

**Comment by lutzarno** [comment]
I would like to be able to select whether the average value or the median value is used for a distribution graph on the statistics page. 
With the average as separating line values tend to accumulate on the left side of that line.

**Comment by NobodyForNothing** [comment]
@lutzarno I'm not quite sure I fully understood your idea. The average text is always displayed at the horizontal center of the distribution. If you're talking about the dotted line, it is also positioned independent from the values and the graph should be symmetric to it.

If this is more about seeing the median value, I could display it bellow the average without needing an option 🙂.

**Comment by lutzarno** [comment]
So the average displayed at the horizontal center is actually a mean value? Looks like I mistook the average as just (MAX+MIN)/2. 

I guess the distribution of the values was misleading me. I exported the measurements and calculated their mean and median. The differences are actually minimal (Mean: 120,6/72,8/69,0 Median: 120/73/69).

I still can't understand why the graph looks wrong to me then.

**Comment by NobodyForNothing** [comment]
@lutzarno 

<details><summary>Here is how I understand the terms (for reference):</summary>

### Average / Mean:
- `sum of values / number of values`

### Median:
1. sort all values
2. take the value in the center (`values[values.count / 2]`)

</details>

### The graph

> I still can't understand why the graph looks wrong to me then.

If you are using an app version before v1.7.1 there was a calculation error in the graph.

To clarify what behavior I expect (this should probably be explained in the app): The graph in the value distribution counts how often you saved the corresponding value. 

For example: take the list of sys measurements `123, 122, 120, 123` will result in the following bars and relative heights `120(1), 121(0), 122(1), 123(2)`.

**Comment by NobodyForNothing** [comment]
I would prefer your other solutions like allowing to ignore the time submitted by the device or automatically updating the clock. Two timestamps would be a little unexpected and would need some explanation.

Does the `lifetime` interval miss any measurements for you?

**Comment by Ilmanikusc29** [comment]
Hello,
some guidelines recommend taking multiple consecutive blood pressure measurements and then calculating their average. In this approach, only the mean value is stored.

Could this feature be added e.g., as a configurable option?

And does this suggestion fits into this thread or should I create a new one for it?

**Comment by derdilla** [comment]
@Ilmanikusc29 thanks, valid suggestion. Adding a setting seems reasonable. A new issue would help, statistics are unrelated.

How would an intuitive UI look like? Should the setting add an "average with" button to the input or did you have something else in mind?

**Comment by Ilmanikusc29** [comment]
Thank you, I've created a new thread, see https://github.com/derdilla/blood-pressure-monitor-fl/issues/676
